### PR TITLE
Split Filename Template

### PR DESCRIPTION
**Pull request by LePips**
- Closes https://github.com/CreateAPI/CreateAPI/issues/60

Entities and paths now simply have their own `filenameTemplate` config option. This does not affect extension files. The original `--filename-template` flag was removed.

### CODE REVIEW

File: Tests/CreateAPITests/GenerateOptionsTests.swift [hunk 1]
```diff
@@ -657,4 +641,26 @@ final class GenerateOptionsTests: GenerateBaseTests {
...
         try compare(package: "petstore-identifiable")
     }
+    
+    func testPetstoreFilenameTemplate() throws {
+        // GIVEN
+        let command = try Generate.parse([
+            pathForSpec(named: "petstore", ext: "yaml"),
+            "--output", temp.url.path,
+            "--package", "petstore-filename-template",
+            "--generate", "entities",
```
Comment (liamnichols) [inline Tests/CreateAPITests/GenerateOptionsTests.swift:651]: Can we test generating both?

File: Tests/CreateAPITests/GenerateOptionsTests.swift [hunk 2]
```diff
@@ -18,22 +18,6 @@ final class GenerateOptionsTests: GenerateBaseTests {
...
-            "--output", temp.url.path,
-            "--package", "petstore-change-filename",
-            "--filename-template", "%0.generated.swift"
-        ])
-        
-        // WHEN
-        try command.run()
-        
-        // THEN
-        try compare(package: "petstore-change-filename")
```
Comment (liamnichols) [inline Tests/CreateAPITests/GenerateOptionsTests.swift:34]: Please be sure to remove **Expected/petstore-change-filename** as well as the dependency from in **AllPackages/Package.swift**

Comment (LePips) [reply Tests/CreateAPITests/GenerateOptionsTests.swift:34]: Is there a way we can automate the population of that file?

File: Tests/CreateAPITests/GenerateOptionsTests.swift (hunk 1 above)
Comment (LePips) [reply Tests/CreateAPITests/GenerateOptionsTests.swift:651]: Whoops, definitely should.

File: Tests/CreateAPITests/GenerateOptionsTests.swift (hunk 2 above)
Comment (liamnichols) [reply Tests/CreateAPITests/GenerateOptionsTests.swift:34]: In theory I think so, but its a little tricky so I don't have an exact answer as to how though

**Pull request merged**